docs/root/api/starting_envoy.rst

```diff
@@ -202,6 +202,21 @@ for further information.
   // Swift
   builder.addPerTryIdleTimeoutSeconds(5)
 
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+``addH2StreamBufferLimitBytes``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Specifies the limit in bytes that the engine will buffer for h2 streams when using Envoy filters
```

Comment:
Actually sorry coming back to this one more time, I think this doesn't work.
you're trying to set the buffer limits for what the HCM will buffer.
The HCM buffers based on the downstream "limits" which for H1 is connection limit and H2 is stream limit.  But for E-M, Downstream isn't H1 or H2 it's our custom library.  It looks like the limit for the DirectStream is hard-coded?

uint32_t bufferLimit() override { return 65000; }

Reply:
@alyssawilk that makes sense. But now I am confused about what the config I am setting here does then?

Reply:
I believe you're changing what the buffer limit would be if we had an HTTP/2 downstream codec instead of the Envoy Mobile codec. Happy to sync up on slack if it'd help

I think to move forward here, you're going to want something like
https://github.com/envoyproxy/envoy/blob/main/test/integration/filters/stream_info_to_headers_filter.cc
for test/common/integration/client_integration_test.cc

It basically lets you convert things available to filters and stream info into headers.  In this case you could log the filter's buffer limits as a header, and confirm my suspicion that tweaking the H2 downstream option doesn't actually change the limits derived from the hard-coded Envoy-Mobile codec.

Reply:
Thanksgiving got in the way of this change. I will pick back up this week.

Reply:
Thanks for all the help @alyssawilk